### ISSUE TITLE
imgmount does not accept relative paths within a mounted system

### ISSUE THREAD

**Issue by Nazosan** [joncampbell123/dosbox-x#3617]
### Describe the bug

This has been a thing that has confused me several times about DOSBox-X specifically (I am pretty sure I remember the original DOSBox not doing this.)  When in a mounted system -- say a mounted C: folder or the automatic drivez mount -- if you use imgmount without specifying an absolute path, it produces an error that can be quite confusing saying the file is not found.  I'm not clear on if this is 100% intentional behavior or not, but it 

### Steps to reproduce the behaviour

1.  Start DOSBox-X
2.  Use a mounted filesystem such as drive Z: or any other (say, for example, "C: then cd \game")
3.  Type imgmount [letter] [filename] where [filename] is a file that does exist within that directory.

### Expected behavior

imgmount should accept a relative path within the mounted filesystem

### What operating system(s) this bug have occurred on?

Probably all OSes, but certainly all versions of Windows.  I am running 10, but I believe it's the same for all of them.

### What version(s) of DOSBox-X have this bug?

All recent builds, including the most current (0.84.1)

### Output log

```text
C:\GAME>imgmount x existingfile.example
Image file not found.
Must specify image file(s) to mount.
```

### Additional information

I'm sure this has come up before and perhaps it is intentional behavior.  At a very minimum there should probably be a command line parameter to override this behavior (say -r for relative?)  However, I think it is generally better for intuitiveness and convenience if it is relative by default and would require a parameter instead to override that (say -h for host?)  There are a lot of disadvantages to requiring a full absolute path including the simple possibility of it being run on a different computer (ex from a USB drive) where drive letters may be different.  While my own usage may differ (I like to have a relatively universal DOSBox-X setup and keep my backup images in the same folder as the installed game so I can just have a simple batch file that mounts, runs the game, then unmounts, but I know a lot of people actually use a bunch of separate config files) I do know that there are a number of people running into the same unintuitive error and being confused by it by simple virtue of the fact I've had to search this several times due to forgetting and thankfully finding the answer where others have asked.

Even if the command itself is not changed it should definitely produce a slightly more intuitive error message or maybe even explicitly state something like "full host path required, local file ignored" or something like that so the user at least knows what is going on without having to search online for the explanation.

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by rderooy** [comment]
I **think** imgmount the rational is simply that you are much more likely to mount an image from the host filesystem then from a filesystem inside dosbox. As such it makes perfect sense to me and I would hate for it to change.

**Comment by Nazosan** [comment]
I'm not sure how many people would feel the same way, but, I did suggest a command line parameter.

I think though, for starters, if a path is not actually specified it would make sense for it to at least search the current location.  If it comes to that, it can search both (that's only two paths to check anyway after all.)  You can even give preference to the host's current path rather than the mounted system's current path (eg check that first and if the filename specified is found there use that without checking the mounted path then.)  If the user isn't writing out a full length path name, they probably mean "here."

**Comment by maron2000** [comment]
You guys might want to take it easy.

`MOUNT` or `IMGMOUNT` DOES accept both relative and full-path designation.
For instance, if you `mount d .` then you mount your work directory which is the current directory.

The work directory can be specified by `working directory option` in the conf file.
If you want to make it where the DOSbox-X executable is, set it to `program`

You can explicitly set the work directory to other directory as well.
Details are mentioned in the conf file.

**Comment by Nazosan** [comment]
Then let me change that to "does not accept current relative directory" because if you try, say, "imgmount x filename.xxx" for a file in the directory you are currently changed into in the emulated machine it says the file does not exist and you have to type in a full pathname still.  Honestly I guess at the very simplest, if basically nothing else is changed, it should at least search both in that particular case (and you can give priority to the host's current directory if you prefer.)  It's also still weird that the error message misleadingly tells you that you must specify an image as if you've typed the command line wrong or are missing a parameter.  If noting else is changed, maybe it could at least say something less confusing like "file not found in main working directory" or something like that so the user can be more clear what the real error is.

**Comment by maron2000** [comment]
I can mount image files on a mounted DOSBox-X drive flawlessly.
Relative directory is converted to full-path of your host automatically by DOSBox-X.
Is this the case you're explaining?

![command_001](https://user-images.githubusercontent.com/68574602/178100019-4a832ef5-0594-4461-8d60-c61d0e2b60d1.png)

**Comment by Nazosan** [comment]
Ok, this is interesting and I apologize for insufficient testing.  This _was_ a problem in the past (I specifically remember running into it trying to mount my Warcraft 2 backup image in particular) but it does seem to work normally when I try to run that game today.  _However_, it seems there are still conditions in which it does not work.  In particular I know drive Z seems to produce that error.  I thought I'd try these to see what the fuss was about:
![command_000](https://user-images.githubusercontent.com/13072654/178122873-d0b928c5-1d1c-4de6-85fa-7051db3a17c0.jpg)
For PC-98 mode everyone uses virtual floppy or harddrive images (which it's generally easiest to mount, then boot) so I found it easiest to put the images on my Z drive rather than mounting a separate drive just for them and this was where I was seeing the error.  Given that it was the same thing as before I admit I assumed nothing had changed since then without checking.  However, if I run it from a user-mounted path, it works:
![command_001](https://user-images.githubusercontent.com/13072654/178123159-75184ff3-c9da-415f-bd11-c65b30674751.jpg)
I don't know if this is intended behavior or not, but it seems that was what was confusing me.

**Comment by maron2000** [comment]
As far as I tried. even if you specify in full-path, images on Z: drive cannot be mounted.
You can use `imagemount d drivez\XXX.img` as a workaround.
(Assuming work directory is where the DOSBox-X executable is)

**Comment by Nazosan** [comment]
Yeah, the only way I got them to work from Z was to use the full host path to the filename (aka not really Z anymore then.)

It's not that big of a deal.  I was trying to do that and thought the old problem still remained.  It does not though.  Either this is a bug or by intent, I'm not sure, but at least it's pretty minor and easy enough to get around.  I just moved the images to a different folder mounted to a different drive letter (using Y for now because why not) and that works fine.  Normally it's easier just to have the images on the C drive anyway along with the games themselves, just since I was trying out the PC-98 emulation, it's tricky with a lot of things basically being hardcoded to expect a specific FDD/IDE layout so images had to be loaded just so and ran into this along the way.

**Comment by color-theory** [comment]
I just ran into this as well and it was incredibly confusing.
